simple-example/src/main.py

```diff
@@ -21,31 +22,36 @@ def main():
     sequence_f1 = os.getenv("SEQUENCE_F1", SEQUENCE_F1_DEFAULT)
     
     # Check each of the values are in an acceptable format for this model
-    try:
-        sequence_length = int( sequence_length )
-    except ValueError:
+    if not is_int(sequence_length) :
         sys.exit("Error: SEQUENCE_LENGTH must be a whole number")
 
-    if sequence_length <= 0:
-        sys.exit("Error: SEQUENCE_LENGTH must be greater than zero")
+    if not int(sequence_length) >= SEQUENCE_LENGTH_MINIMUM:
+        sys.exit(f"Error: SEQUENCE_LENGTH must be a minimum of {SEQUENCE_LENGTH_MINIMUM-1}")
```

Comment:
I like that there are now checks for each of the inputs that provide direct feedback to which variables do not conform to the requirements. I don't like that this now requires `int(variable)` everywhere the value is to be used. These should probably converted to ints a single time once we know it is safe to do so e.g.

```py
if not is_int(sequence_length):
    sys.exit(...)
sequence_length = int(sequence_length)
```

Or wrap that all up in a function as mentioned before.

Reply:
I don't like sys.exit from anything except the main function and I also wanted each message to be clear for those reading it, that's why it's implemented this way. I don't think it needs to be perfect code just good enough for others to understand **why** we are doing each thing. 

I'll move int(sequence_length) for you.

Reply:
Thinking about this further, we could have a Python library that could take the model definition and check all the inputs are valid itself.

But then the question becomes should the model step itself use the model-definition to detect incompatible inputs before it even runs. Maybe even attempt to find a compitable transform.